File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romans = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I']
        numerals = [1000, 900, 500, 400, 100, 90, 50, 40 ,10 ,9 ,5, 4, 1]

        num = 0
        count = 0
        while s or count == 20:
            count += 1
            print(s, num)
            single = s[0]
            if len(s) > 1:
                double = s[0] + s[1]
            else:
                double = ''

            for i in range(len(romans)):
                if romans[i] == double:
                    num += numerals[i]
                    s = s.replace(romans[i], '', 1)
                    break

                elif romans[i] == single:
                    num += numerals[i]
                    s = s.replace(romans[i], '', 1)
                    break

        return num

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int num = 0;
        char next = 'a';
        boolean doub = false;

        for(int i = 0; i < s.length(); i++){
            if(doub == true){
                doub = false;
                continue;
            }
            System.out.print(s.charAt(i));

            if(i + 1 < s.length()){
                next = s.charAt(i + 1);
            }
            
            System.out.println(next);

            String checker = "" + s.charAt(i) + next;
            if (checker.equals("IV")){
                num += 4;
                doub = true;
                //System.out.println("boop");
                continue;

            }
            else if (checker.equals("IX")){
                num += 9;
                doub = true;
                continue;
            }
            
            else if (checker.equals("IX")){
                num += 9;
                doub = true;
                continue;
            }
            
            else if (checker.equals("XL")){
                num += 40;
                doub = true;
                continue;
            }
            
            else if (checker.equals("XC")){
                num += 90;
                doub = true;
                continue;
            }
            
            else if (checker.equals("CD")){
                num += 400;
                doub = true;
                continue;
            }
            
            else if (checker.equals("CM")){
                num += 900;
                doub = true;
                continue;
            }


            switch(s.charAt(i)){
                case 'I':
                    num += 1;
                    break;
                case 'V':
                    num += 5;
                    break;
                case 'X':
                    num += 10;
                    break;
                case 'L':
                    num += 50;
                    break;
                case 'C':
                    num += 100;
                    break;
                case 'D':
                    num += 500;
                    break;
                case 'M':
                    num += 1000;
                    break;

            }
        }

        return num;
    }
}